0022-circuit-block.md

```diff
@@ -0,0 +1,300 @@
+include "stdgates.inc";
+
+@ibm twirl
+box {
+    cx $0, $1;
+}
+```
+
+Certain hardware providers may also have direct-access APIs that do not require submission of the job via Qiskit.
+Custom serialisers/deserialisers of annotations should be aware of this as well; it is effectively the same problem as serialisation to and from OpenQASM 3.
+
+
+### Communication of "allowed" annotations from a backend
+
+For hardware instructions, we use the `Target` to represent the ISA of a given QPU.
+A `box` annotation remaining on a circuit after Qiskit SDK transpilation implies that the backend will do further processing on the circuit to prepare this for execution.
+This can be associated with further processing by the "quantum computer" before final compilation to the QPU.
+
+In IBM's specified terminology, a "quantum computer" is the full execution environment that a user interacts with to submit quantum jobs to, and might perform further circuit processing, error mitigation, etc, as part of passing the circuit down to the control hardware.
+A "QPU" is more of a representation of the control hardware itself.
+
+In Qiskit SDK, a `Target` represents a QPU, while we might (but have not necessarily yet done so) associate the `Backend` with what we call the entire "quantum computer".
+As an annotation implies further processing by the quantum computer before execution by the QPU, the information on what annotations may be left in the circuit by the transpiler should be drawn from new information specified by the quantum computer, not the particular QPU.
+
+The transpiler should fail if a user or a pass in its pass manager inserts annotations on boxes that the backend does not support, and are not removed by the end of the transpilation.
+We need a way to communicate this information from backends to Qiskit SDK.
+
+For example, an advanced user might attempt to compile an `Estimator` pub with explicit IBM-specific markings of how the noise learning should be performed.
+The compilation of a pub represents partial compilation for a particular quantum computer, which will in turn perform further processing to turn this into the set of circuits that will actually be run on the QPU, and so Qiskit SDK's transpiler perhaps ought to have a similar specification of what constitutes "valid" output of such a compilation for quantum-computer processing, much like `Target` represents what constitues valid output for QPU processing.
+
+#### Very very rough sketch of how this might motivate a `BackendV3`
+
+> [!NOTE]
+> This is a _very vague_ sketch, I'm just trying to make sure we have a clear definition of what a `Target` is, and we don't start having it become so abstract that it is no longer a good model for a QPU.
+> This is absolutely not necessary to finalise for this RFC, but I feel like it's worth thinking about this as a joined-up story, since previously we have treated the output of `transpile` as "ready for QPU execution", but annotations stymie that.
```

Comment:
There are other things that stymie that too at the moment, like parameter expressions, which the quantum computer handles but the QPU doesn't. All in all, I really like the direction this section is going in terms of making the distinction between QPU<->Target, Quantum Computer<->Backend nice and crisp.

As for the `sample` and `estimate` methods on backend, I think my biggest concern here is whiplash: we just yoinked `run` away, and this would look like backpedaling. It seems like we could alternately make primitives V3 have to own a backend V3 instance.